File: Implementation/LV2_43165.cpp
#include <string>
#include <vector>
#include <iostream>

// 8:05 ~ 8:14

using namespace std;

int N, T;
int count = 0;
vector <int> tree;

void makeTarget(int pos, int num) {
    
    // 끝까지 돌았을 때
    if (pos == N) {
        if (num == T) count++;
        return ;
    }
    
    // 빼기
    makeTarget(pos+1, num + tree[pos]);
    
    // 더하기
    makeTarget(pos+1, num - tree[pos]);
}

int solution(vector<int> numbers, int target) {
    int answer = 0;
    
    // 전역변수
    N = numbers.size();
    T = target;
    tree = numbers;
    
    // 메인 함수
    makeTarget(0, 0);
    
    answer = count;
    return answer;
}

File: Graph/LV2_43165.py
answer = 0

def dfs(numbers, index, target):
    global answer
    if index == len(numbers) and target == 0:
        answer += 1
        return
    elif index == len(numbers) and target != 0:
        return 
    else:
        dfs(numbers, index+1, target-numbers[index])
        dfs(numbers, index+1, target+numbers[index])

def solution(numbers, target):
    global answer
    dfs(numbers, 0, target)
    return answer